atomic_reactor/plugins/pre_fetch_sources.py

```diff
@@ -115,6 +115,17 @@ def set_koji_image_build_data(self):
                 )
             raise ValueError(err_msg)
 
+        build_type = self.session.getBuildType(self.koji_build['build_id'], strict=True)
+        if 'image' not in build_type:
+            type_string = list(build_type.keys())[0]
+            err_msg = ('koji build {} is {} build, source container '
+                       'build needs image build'.format(self.koji_build['nvr'], type_string))
+            raise ValueError(err_msg)
+        elif self.koji_build['name'].endswith('-source'):
```

Comment:
notpick: we can do this check first to avoid the koji call. Also, do we need the koji call at all? Couldn't we infer the build type with the ['extra']['image'] or ['extra']['typeinfo']['image'] key presence?

Reply:
changed the check